From f5bfa9dc0c80b3715fa8526d81694aa5c54a1fa5 Mon Sep 17 00:00:00 2001
Message-ID: <f5bfa9dc0c80b3715fa8526d81694aa5c54a1fa5.1780689448.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Fri, 5 Jun 2026 21:55:08 +0200
Subject: [PATCH] remove usage of typing.ByteString

typing.ByteString is deprecated in python 3.15 and slated for removal in
python 3.17. This creates deprecation warnings which trip up our tests.

Since we are requiring python >= 3.10 already, we can switch to the
proper alterantive unconditionally now.
---
 src/__init__.py        | 7 +------
 src/utils.py           | 9 ++-------
 src_classic/fitz_old.i | 7 +------
 src_classic/utils.py   | 7 +------
 4 files changed, 5 insertions(+), 25 deletions(-)

diff --git a/src/__init__.py b/src/__init__.py
index 8b756ccf..2bff8ec0 100644
--- a/src/__init__.py
+++ b/src/__init__.py
@@ -462,12 +462,7 @@ def _format_g(value, *, fmt='%g'):
         
 format_g = _format_g
 
-# ByteString is gone from typing in 3.14.
-# collections.abc.Buffer available from 3.12 only
-try:
-    ByteString = typing.ByteString
-except AttributeError:
-    ByteString = bytes | bytearray | memoryview
+ByteString = bytes | bytearray | memoryview
 
 # Names required by class method typing annotations.
 OptBytes = typing.Optional[ByteString]
diff --git a/src/utils.py b/src/utils.py
index 7d3271ea..e42a1c0b 100644
--- a/src/utils.py
+++ b/src/utils.py
@@ -24,13 +24,8 @@ rect_like = "rect_like"
 matrix_like = "matrix_like"
 quad_like = "quad_like"
 
-# ByteString is gone from typing in 3.14.
-# collections.abc.Buffer available from 3.12 only
-try:
-    ByteString = typing.ByteString
-except AttributeError:
-    # pylint: disable=unsupported-binary-operation
-    ByteString = bytes | bytearray | memoryview
+# pylint: disable=unsupported-binary-operation
+ByteString = bytes | bytearray | memoryview
 
 AnyType = typing.Any
 OptInt = typing.Union[int, None]
diff --git a/src_classic/fitz_old.i b/src_classic/fitz_old.i
index 8abc8955..3e94b3bc 100644
--- a/src_classic/fitz_old.i
+++ b/src_classic/fitz_old.i
@@ -322,12 +322,7 @@ rect_like = "rect_like"
 matrix_like = "matrix_like"
 quad_like = "quad_like"
 
-# ByteString is gone from typing in 3.14.
-# collections.abc.Buffer available from 3.12 only
-try:
-    ByteString = typing.ByteString
-except AttributeError:
-    ByteString = bytes | bytearray | memoryview
+ByteString = bytes | bytearray | memoryview
 
 AnyType = typing.Any
 OptInt = typing.Union[int, None]
diff --git a/src_classic/utils.py b/src_classic/utils.py
index c1369afb..3639834c 100644
--- a/src_classic/utils.py
+++ b/src_classic/utils.py
@@ -24,12 +24,7 @@ rect_like = "rect_like"
 matrix_like = "matrix_like"
 quad_like = "quad_like"
 
-# ByteString is gone from typing in 3.14.
-# collections.abc.Buffer available from 3.12 only
-try:
-    ByteString = typing.ByteString
-except AttributeError:
-    ByteString = bytes | bytearray | memoryview
+ByteString = bytes | bytearray | memoryview
 
 AnyType = typing.Any
 OptInt = typing.Union[int, None]
-- 
2.54.0.1146.g083d50a523

